### ISSUE TITLE
For processing some large image, 'ResizingImageProcessor' is much better than 'DownsmaplingProcessor''

### ISSUE THREAD

**Issue by czeludzki** [onevcat/Kingfisher#2091]
It is not strange, I can expect this behavior:

1. Kingfisher has a default strategy that not using more than 1/4 of the device memory for memory caching purposes.
2. Any newly added images to memory cache would trigger a purge for old images.
3. So as long as you are loading more images in a deeper view controller, the previous images will be removed from Kingfisher's memory cache.
4. However, these images are still there in older view controllers. So they are not released.
5. So the new images can only be loaded from disk, and rendered again.

It causes this unbounded memory increment. In this specific case, to avoid that, maybe you need to consider set the `image` in image view back to `nil` in `viewDidDisapper` in old view controllers.

> As far as I understand, this is kind of expected result and the only option is to avoid . aspectFill for similar view hierarchy?

- Maybe just set the `contentMode` in the image view is good enough for your case (if you are loading such images in similar size as your sample). This would avoid the raw pixel redrawing and gives a better result.
- If you have very large images, I suggest trying the `DownsamplingImageProcessor`.
- If you have an unlimited view controller stack, please reset the `image` in image view back to `nil` in `viewDidDisappear`, and set it again in `viewWillAppear`.

_Originally posted by @onevcat in https://github.com/onevcat/Kingfisher/issues/1506#issuecomment-691122426_

I have some large images need to show(size is about {width: 100000 height: 100000}). 

I using the CATiledLayer support, on most of devices it running fine(like iPhone x and latest). But on a iPhone6s, iOS13, low battery health(80%), the memory consumption is not optimistic.

For showing the large image, I referred to Apple's CATiledLayer Demo: https://developer.apple.com/library/archive/samplecode/LargeImageDownsizing. 
Roughly, I cover two layer of views on the screen, one of it is UIImageView, other is a CATiledLayer wrapped. UIImageView is for showing a small size image from the original image, and it cover by the CATiledLayer wrapped. The small size image is from the 'ResizeingImageProcessor'.

For optimizing on that iPhone6s device, I tried your suggesstion, using the 'DownsamplingImageProcessor', but it's getting worse...
It cost much more memory and let the main thread become not smoothly.

In conclusion, 'DownsmaplingImageProcessor' is not like your describe, may be it is unsuitable to using in the large image scene.

**Comment by BhavinBhadani** [comment]
@czeludzki I have a same issue. Do you have any solutions for this?

**Comment by czeludzki** [comment]
> @czeludzki我有同样的问题。您对此有什么解决方案吗？

At last, based on the results obtained through actual observation, I chose the ResizingImageProcessor(referenceSize: UIScreen.main.bounds.size, mode: .aspectFit) processor. This achieved the desired effect for me, which is to process large images into thumbnails without consuming a lot of RAM memory.

Additionally, you can clone my project and replace the ResizingImageProcessor with the DownsamplingProcessor to see the effects.
https://github.com/czeludzki/CZImagePreviewer/blob/dd7855eca2910e2d92e44a055d85c7d15021b312/CZImagePreviewer/Classes/TiledImageViewWrapper.swift#L57

**Comment by BhavinBhadani** [comment]
Thanks @czeludzki . Will check that

**Comment by BhavinBhadani** [comment]
@czeludzki Image loading from URL takes lot of time load. Do you have any idea about it? Also memory consumption is little bit manageable now but still loading time is too long

**Comment by czeludzki** [comment]
> @czeludzki Image loading from URL takes lot of time load. Do you have any idea about it? Also memory consumption is little bit manageable now but still loading time is too long

@BhavinBhadani The user's network environment is uncontrollable, and I have no idea about it. Perhaps providing UI prompts to users during the waiting period is the best effort we can make.
